scripts/ci/pre_commit/check_providers_subpackages_all_have_init.py

```diff
@@ -30,40 +32,119 @@
     "static",
 ]
 
+PATH_EXTENSION_STRING = '__path__ = __import__("pkgutil").extend_path(__path__, __name__)  # type: ignore'
+
+# Here we should add the second level paths that we want to have sub-packages in
+KNOWN_SECOND_LEVEL_PATHS = ["apache", "atlassian", "common", "cncf", "dbt", "microsoft"]
```

Comment:
This list is in a lot of places. If we ever have to add to it, it will be tricky to find them all :grimacing:

Reply:
Yes. I plan to find a way to derive it automatically in a follow-up.  But this PR is already too large to try to do it now :)